### PR TITLE
User authentication

### PR DESCRIPTION
**Pull request by RobynVG**
References #19 

Provides user authentication for bookstore owner or User. Set up with one user and one admin with credentials "user","user" and "admin","admin".

To test pull request:
Attempt to reach endpoints without authenticating
Ensure admin can upload and user cannot

Clearing the browser's cache is the easiest way to switch users

### CODE REVIEW

File: src/main/java/com/kyhorne/bookstore/BookstoreApplication.java [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 
 @SpringBootApplication
 public class BookstoreApplication {
 	public static void main(String[] args) {
 		SpringApplication.run(BookstoreApplication.class, args);
 	}
+
+	//@FIXME Need appropriate place to add default admin
```
Comment (devonverge) [inline src/main/java/com/kyhorne/bookstore/BookstoreApplication.java]: Do we still need to add default admin location?

File: src/main/java/com/kyhorne/bookstore/configurations/SpringSecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            .jdbcAuthentication();
+    }
+
+    // Secure the endpoints with HTTP authentication
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .authorizeRequests()
+            .mvcMatchers("/api/v1/upload").hasAnyRole("ADMIN")
+            //FIXME provides default browser login form but user needs to be able to register
```
Comment (devonverge) [inline src/main/java/com/kyhorne/bookstore/configurations/SpringSecurityConfiguration.java]: Do we still need to fix this?

File: src/main/java/com/kyhorne/bookstore/BookstoreApplication.java (hunk 1 above)
Comment (RobynVG) [reply src/main/java/com/kyhorne/bookstore/BookstoreApplication.java]: If we don't hard code an admin we would have to allow the admin to register. Which we can do but I can't think of a way to do that. 
By the comment I meant that I'm not sure if BookstoreApplication.java is the appropriate place for that code

File: src/main/java/com/kyhorne/bookstore/configurations/SpringSecurityConfiguration.java (hunk 2 above)
Comment (RobynVG) [reply src/main/java/com/kyhorne/bookstore/configurations/SpringSecurityConfiguration.java]: Yes and no. It works but we will need to add a rule that allows a user access to the "login or register" page without authenticating

**Pull request merged**